src/main/java/seedu/duke/ui/UI.java

```diff
@@ -50,6 +50,8 @@ public void printMessage(String message) {
      * @return String of input.
      */
     public String readCommand() {
+        // assert that user is entering input with "\n" and not signals like Ctrl+D
+        assert scanner.hasNextLine();
```

Comment:
Oh didn't realise that Crtl-D would cause it to crash. Maybe we can do something to fix this.